File: src/main/java/core/db/VereinTable.java
package core.db;

import core.model.misc.Verein;
import core.util.HOLogger;

import java.sql.ResultSet;
import java.sql.Types;

final class VereinTable extends AbstractTable {
	final static String TABLENAME = "VEREIN";

	protected VereinTable(JDBCAdapter  adapter){
		super( TABLENAME, adapter );
	}

	@Override
	protected void initColumns() {
		columns 	= new ColumnDescriptor[16];
		columns[0]	= new ColumnDescriptor( "HRF_ID",		Types.INTEGER,false, true );
		columns[1]	= new ColumnDescriptor( "COTrainer",		Types.INTEGER,false );
		columns[2]	= new ColumnDescriptor( "Pschyologen",	Types.INTEGER,false );
		columns[3]	= new ColumnDescriptor( "Finanzberater", Types.INTEGER,false );
		columns[4]	= new ColumnDescriptor( "PRManager",		Types.INTEGER,false );
		columns[5]	= new ColumnDescriptor( "Aerzte",		Types.INTEGER,false );
		columns[6]	= new ColumnDescriptor( "Jugend",		Types.INTEGER,false );
		columns[7]	= new ColumnDescriptor( "Siege",			Types.INTEGER,false );
		columns[8]	= new ColumnDescriptor( "Ungeschlagen",	Types.INTEGER,false );
		columns[9]	= new ColumnDescriptor( "Fans",			Types.INTEGER,false );
		columns[10]	= new ColumnDescriptor( "TacticAssist",	Types.INTEGER,false );
		columns[11]	= new ColumnDescriptor( "FormAssist",	Types.INTEGER,false );
		columns[12]	= new ColumnDescriptor( "GlobalRanking",	Types.INTEGER,false );
		columns[13]	= new ColumnDescriptor( "LeagueRanking",	Types.INTEGER,false );
		columns[14]	= new ColumnDescriptor( "RegionRanking",	Types.INTEGER,false );
		columns[15]	= new ColumnDescriptor( "PowerRating",	Types.INTEGER,false );
	}

	/**
	 * speichert das Verein
	 */
	void saveVerein(int hrfId, Verein verein) {
		String statement ;
		final String[] awhereS = { "HRF_ID" };
		final String[] awhereV = { "" + hrfId };

		if (verein != null) {
			//first delete existing entry
			executePreparedDelete( hrfId);
			executePreparedInsert(
					hrfId,
					verein.getCoTrainer(),
					verein.getPsychologen(),
					verein.getFinancialDirectorLevels(),
					verein.getPRManager(),
					verein.getAerzte(),
					verein.getJugend(),
					verein.getSiege(),
					verein.getUngeschlagen(),
					verein.getFans(),
					verein.getTacticalAssistantLevels(),
					verein.getFormCoachLevels(),
					verein.getGlobalRanking(),
					verein.getLeagueRanking(),
					verein.getRegionRanking(),
					verein.getPowerRating()
			);
	}
	}

	@Override
	protected PreparedSelectStatementBuilder createPreparedSelectStatementBuilder(){
		return new PreparedSelectStatementBuilder(this, "WHERE HRF_ID=?");
	}
	/**
	 * lädt die Basics zum angegeben HRF file ein
	 */
	Verein getVerein(int hrfID) {
		Verein club = new Verein();

		if (hrfID != -1) {
			ResultSet rs = executePreparedSelect(hrfID);
			try {
				if (rs != null) {
					rs.next();
					club = new Verein(rs);
				}
			} catch (Exception e) {
				HOLogger.instance().log(getClass(), "DatenbankZugriff.getTeam: " + e);
			}
		}

		return club;
	}

}

File: src/main/java/core/db/VereinTable.kt
package core.db

import core.model.misc.Verein
import java.sql.*
import java.util.function.BiConsumer
import java.util.function.Function

internal class VereinTable(adapter: JDBCAdapter) : AbstractTable(TABLENAME, adapter) {
    override fun initColumns() {
        columns = arrayOf<ColumnDescriptor>(
            ColumnDescriptor.Builder.Companion.newInstance().setColumnName("HRF_ID")
                .setGetter(Function<Any?, Any?>({ p: Any? -> (p as Verein?)!!.getHrfId() })).setSetter(
                BiConsumer<Any?, Any>({ p: Any?, v: Any -> (p as Verein?)!!.setHrfId(v as Int) })
            ).setType(Types.INTEGER).isPrimaryKey(true).isNullable(false).build(),
            ColumnDescriptor.Builder.Companion.newInstance().setColumnName("COTrainer")
                .setGetter(Function<Any?, Any?>({ p: Any? -> (p as Verein?)!!.getCoTrainer() })).setSetter(
                BiConsumer<Any?, Any>({ p: Any?, v: Any -> (p as Verein?)!!.setCoTrainer(v as Int) })
            ).setType(Types.INTEGER).isNullable(false).build(),
            ColumnDescriptor.Builder.Companion.newInstance().setColumnName("Pschyologen")
                .setGetter(Function<Any?, Any?>({ p: Any? -> (p as Verein?)!!.getPsychologen() })).setSetter(
                BiConsumer<Any?, Any>({ p: Any?, v: Any -> (p as Verein?)!!.setPsychologen(v as Int) })
            ).setType(Types.INTEGER).isNullable(false).build(),
            ColumnDescriptor.Builder.Companion.newInstance().setColumnName("Finanzberater")
                .setGetter(Function<Any?, Any?>({ p: Any? -> (p as Verein?)!!.getFinancialDirectorLevels() }))
                .setSetter(
                    BiConsumer<Any?, Any>({ p: Any?, v: Any -> (p as Verein?)!!.setFinancialDirectorLevels(v as Int) })
                ).setType(
                Types.INTEGER
            ).isNullable(false).build(),
            ColumnDescriptor.Builder.Companion.newInstance().setColumnName("PRManager")
                .setGetter(Function<Any?, Any?>({ p: Any? -> (p as Verein?)!!.getPRManager() })).setSetter(
                BiConsumer<Any?, Any>({ p: Any?, v: Any -> (p as Verein?)!!.setPRManager(v as Int) })
            ).setType(Types.INTEGER).isNullable(false).build(),
            ColumnDescriptor.Builder.Companion.newInstance().setColumnName("Aerzte")
                .setGetter(Function<Any?, Any?>({ p: Any? -> (p as Verein?)!!.getAerzte() })).setSetter(
                BiConsumer<Any?, Any>({ p: Any?, v: Any -> (p as Verein?)!!.setAerzte(v as Int) })
            ).setType(Types.INTEGER).isNullable(false).build(),
            ColumnDescriptor.Builder.Companion.newInstance().setColumnName("Jugend")
                .setGetter(Function<Any?, Any?>({ p: Any? -> (p as Verein?)!!.getJugend() })).setSetter(
                BiConsumer<Any?, Any>({ p: Any?, v: Any -> (p as Verein?)!!.setJugend(v as Int) })
            ).setType(Types.INTEGER).isNullable(false).build(),
            ColumnDescriptor.Builder.Companion.newInstance().setColumnName("Siege")
                .setGetter(Function<Any?, Any?>({ p: Any? -> (p as Verein?)!!.getSiege() })).setSetter(
                BiConsumer<Any?, Any>({ p: Any?, v: Any -> (p as Verein?)!!.setSiege(v as Int) })
            ).setType(Types.INTEGER).isNullable(false).build(),
            ColumnDescriptor.Builder.Companion.newInstance().setColumnName("Ungeschlagen")
                .setGetter(Function<Any?, Any?>({ p: Any? -> (p as Verein?)!!.getUngeschlagen() })).setSetter(
                BiConsumer<Any?, Any>({ p: Any?, v: Any -> (p as Verein?)!!.setUngeschlagen(v as Int) })
            ).setType(Types.INTEGER).isNullable(false).build(),
            ColumnDescriptor.Builder.Companion.newInstance().setColumnName("Fans")
                .setGetter(Function<Any?, Any?>({ p: Any? -> (p as Verein?)!!.getFans() })).setSetter(
                BiConsumer<Any?, Any>({ p: Any?, v: Any -> (p as Verein?)!!.setFans(v as Int) })
            ).setType(Types.INTEGER).isNullable(false).build(),
            ColumnDescriptor.Builder.Companion.newInstance().setColumnName("TacticAssist")
                .setGetter(Function<Any?, Any?>({ p: Any? -> (p as Verein?)!!.getTacticalAssistantLevels() }))
                .setSetter(
                    BiConsumer<Any?, Any>({ p: Any?, v: Any -> (p as Verein?)!!.setTacticalAssistantLevels(v as Int) })
                ).setType(
                Types.INTEGER
            ).isNullable(false).build(),
            ColumnDescriptor.Builder.Companion.newInstance().setColumnName("FormAssist")
                .setGetter(Function<Any?, Any?>({ p: Any? -> (p as Verein?)!!.getFormCoachLevels() })).setSetter(
                BiConsumer<Any?, Any>({ p: Any?, v: Any -> (p as Verein?)!!.setFormCoachLevels(v as Int) })
            ).setType(Types.INTEGER).isNullable(false).build(),
            ColumnDescriptor.Builder.Companion.newInstance().setColumnName("GlobalRanking")
                .setGetter(Function<Any?, Any?>({ p: Any? -> (p as Verein?)!!.getGlobalRanking() })).setSetter(
                BiConsumer<Any?, Any>({ p: Any?, v: Any -> (p as Verein?)!!.setGlobalRanking(v as Int) })
            ).setType(Types.INTEGER).isNullable(false).build(),
            ColumnDescriptor.Builder.Companion.newInstance().setColumnName("LeagueRanking")
                .setGetter(Function<Any?, Any?>({ p: Any? -> (p as Verein?)!!.getLeagueRanking() })).setSetter(
                BiConsumer<Any?, Any>({ p: Any?, v: Any -> (p as Verein?)!!.setLeagueRanking(v as Int) })
            ).setType(Types.INTEGER).isNullable(false).build(),
            ColumnDescriptor.Builder.Companion.newInstance().setColumnName("RegionRanking")
                .setGetter(Function<Any?, Any?>({ p: Any? -> (p as Verein?)!!.getRegionRanking() })).setSetter(
                BiConsumer<Any?, Any>({ p: Any?, v: Any -> (p as Verein?)!!.setRegionRanking(v as Int) })
            ).setType(Types.INTEGER).isNullable(false).build(),
            ColumnDescriptor.Builder.Companion.newInstance().setColumnName("PowerRating")
                .setGetter(Function<Any?, Any?>({ p: Any? -> (p as Verein?)!!.getPowerRating() })).setSetter(
                BiConsumer<Any?, Any>({ p: Any?, v: Any -> (p as Verein?)!!.setPowerRating(v as Int) })
            ).setType(Types.INTEGER).isNullable(false).build()
        )
    }

    /**
     * store team info
     */
    fun saveVerein(hrfId: Int, verein: Verein?) {
        if (verein != null) {
            verein.setHrfId(hrfId)
            verein.stored = isStored(hrfId)
            store(verein)
        }
    }

    /**
     * load team basic information
     */
    fun loadVerein(hrfID: Int): Verein {
        var ret: Verein? = loadOne(Verein::class.java, hrfID)
        if (ret == null) ret = Verein()
        return ret
    }

    companion object {
        val TABLENAME: String = "VEREIN"
    }
}